lib/network_ssl/network_ssl_compat.c

```diff
@@ -4,10 +4,81 @@
 
 #include <openssl/ssl.h>
 
+#ifndef OPENSSL_VERSION_NUMBER
+#error "OPENSSL_VERSION_NUMBER must be defined"
+#endif
+
+/*
+ * LibreSSL claims to be OpenSSL 2.0, but (currently) has APIs compatible with
+ * OpenSSL 1.0.1g.
+ */
+#ifdef LIBRESSL_VERSION_NUMBER
+#undef OPENSSL_VERSION_NUMBER
+#define OPENSSL_VERSION_NUMBER 0x1000107fL
```

Comment:
We might need to tweak this: libressl only added the hostname checking API in 2.7.0 (2018-03-21).  At the moment, we don't `#ifdef` our use of those APIs, so modern libressl will work, but 14-month-old libressl won't.

I've left this unfixed because I think we could add "nicer compiler errors" for that if and when it becomes necessary.  At the moment, I feel like once we have reasonable support for OpenSSL (in particular, the 1.0.2 in Travis-CI), it would be nice to get it merged so that https support isn't held up too much longer.